src/mtev_websocket_client.h

```diff
@@ -24,49 +24,148 @@ typedef struct {
+    \param pool optional, specify an eventer pool; thread will be chosen at random from the pool
+    \param sslconfig optional, enables SSL using the contained config
+    \return boolean indicating success/failure
+    Clients allocated by this function are expected to be interacted with solely through the provided callbacks. There are two guarantees the caller must make:
+    1. The caller must not let a reference to the client escape from the provided callbacks.
+    2. The caller must not call `mtev_websocket_client_free()` with a reference to this client.
+*/
+API_EXPORT(mtev_boolean)
+  mtev_websocket_client_new_noref(const char *host, int port,
+                                  const char *path, const char *service,
+                                  mtev_websocket_client_callbacks *callbacks,
+                                  void *closure,
+                                  eventer_pool_t *pool,
+                                  mtev_hash_table *sslconfig);
+
+/*! \fn void mtev_websocket_client_set_ready_callback(mtev_websocket_client_t *client, mtev_websocket_client_ready_callback ready_callback)
+    \brief Set a new ready_callback on an existing client
+    \param client client to modify
+    \param ready_callback new ready_callback to set
+*/
 API_EXPORT(void)
   mtev_websocket_client_set_ready_callback(mtev_websocket_client_t *client,
-                                           mtev_websocket_client_ready_callback msg_callback);
+                                           mtev_websocket_client_ready_callback ready_callback);
 
+/*! \fn void mtev_websocket_client_set_msg_callback(mtev_websocket_client_t *client, mtev_websocket_client_msg_callback msg_callback)
+    \brief Set a new msg_callback on an existing client
+    \param client client to modify
+    \param msg_callback new msg_callback to set
+*/
 API_EXPORT(void)
   mtev_websocket_client_set_msg_callback(mtev_websocket_client_t *client,
                                          mtev_websocket_client_msg_callback msg_callback);
 
+/*! \fn void mtev_websocket_client_set_cleanup_callback(mtev_websocket_client_t *client, mtev_websocket_client_cleanup_callback cleanup_callback)
+    \brief Set a new cleanup_callback on an existing client
+    \param client client to modify
+    \param cleanup_callback new cleanup_callback to set
+*/
 API_EXPORT(void)
   mtev_websocket_client_set_cleanup_callback(mtev_websocket_client_t *client,
                                              mtev_websocket_client_cleanup_callback cleanup_callback);
 
+/*! \fn mtev_boolean mtev_websocket_client_send(mtev_websocket_client_t *client, int opcode, void *buf, size_t len)
+    \brief Enqueue a message
+    \param client
+    \param opcode opcode as defined in RFC 6455 and referenced in wslay.h
+    \param buf pointer to buffer containing data to send
+    \param len number of bytes of buf to send
+    \return boolean indicating success/failure
+    This function makes a copy of buf of length len.
```

Comment:
Need a blank line between \return and follow-on documentation.. .(here and everywhere)